astro-client-iam-core/client.go

```diff
@@ -39,3 +42,24 @@ func NormalizeAPIError(httpResp *http.Response, body []byte) error {
 	}
 	return nil
 }
+
+func CoreRequestEditor(ctx httpContext.Context, req *http.Request) error {
+	currentCtx, err := context.GetCurrentContext()
+	if err != nil {
+		return nil
+	}
+	os := runtime.GOOS
+	arch := runtime.GOARCH
+	baseURL := currentCtx.GetPublicRESTAPIURL("iam/v1beta1")
+	requestURL, err := url.Parse(baseURL + req.URL.String())
+	if err != nil {
+		return fmt.Errorf("%w, %s", ErrorBaseURL, baseURL)
+	}
+	req.URL = requestURL
+	req.Header.Add("authorization", currentCtx.Token)
+	req.Header.Add("x-astro-client-identifier", "cli")
+	req.Header.Add("x-astro-client-version", version.CurrVersion)
+	req.Header.Add("x-client-os-identifier", os+"-"+arch)
+	req.Header.Add("User-Agent", fmt.Sprintf("astro-cli/%s", version.CurrVersion))
```

Comment:
Should we add a request header for the API version as well? We do want to get to `v1` so tracking API versions in the client could help with this effort. @vandyliu @fredzhy what do you think?

Reply:
API version is already handled on core API side so we don't need that info from the clients